src/borda_clj/client.clj

```diff
@@ -0,0 +1,61 @@
+(ns borda-clj.client
+  (:use [slingshot.slingshot :only [throw+ try+]])
+  (:require [aleph.http :as http]
+            [byte-streams :as bs]
+            [cheshire.core :as json]))
+
+(defn collect
+  "Adds the given measurement (values and dimensions) to the given hashmap and caps the size of the hashmap at the given max-buffer-size"
```

Comment:
I can't think of a better way to put an upper bound on memory consumption, but it's worth thinking what kinds of data distortion can emerge out of `max-buffer-size` semantics.  Some dimension combinations will not get reported at all for some periods, while the ones luckier enough to show up first will get accumulated for all the period. Stats that get reported more frequently are more likely to make it at all.

It occurs to me that it would perhaps be good to have a single stat reporting "how much" data are we leaving out, somehow.  We would need to pre-initialize it at the beginning of every period, since by definition this stat would only get updated when the buffer is full.  If it turns out that we're never hitting the max-buffer-size in practice, this stat would give us confidence in our data.  If we're often leaving out a lot, it would be good to know.